docs/design/README.md

```diff
@@ -0,0 +1,552 @@
+
+Twoliter will eliminate the need for all build-host software other than `docker.`
+A bootstrapping container will be published that contains any software needed to kick off and sustain a `cargo make` build command.
+
+## Kits
+
+A central idea in this design is that the build system will be able to produce and consume Docker images that contain RPM packages.
+We call these images *kits*.
+A kit is a Yum repo distributed in its entirety, atomically, as a container image.
+Kits will satisfy requirements 4-8 above.
+Existing container registry mechanisms, such as ECR, will automatically sign these images satisfying requirement 1.
+
+Builders may reference these kits in order to add packages to their Bottlerocket variants.
+In particular, the core project will publish all of its packages in a suite of kits.
+With these, an out-of-tree maintainer will have all the packages available that go into variants that the core project is currently producing.
+We envision maintainers other than the core team publishing Bottlerocket kits in the future.
+
+### Local and External Kits
+
+During the build process, there will be a distinction between *local kits* and *external kits*.
+A *local kit* is any kit where the packages are defined in the project tree and the maintainer wants to build them from source as part of the Bottlerocket build.
+An *external kit* is a kit that is used by pulling it from a docker registry, i.e. the container image has previously been built.
+
+It should not be a requirement for a maintainer to push a kit to a container registry.
+In other words, it should not be necessary to publish a kit in order to create a Bottlerocket variant.
+This is where *local kits* come in.
+Kits defined in the project tree need not be pre-built before a `twoliter build` command can succeed.
+Twoliter will understand that the local kits need to be built before the rest of the variant build can proceed.
+
+### Kits Project Directory
+
+Currently, there are three directories used for sourcecode when building Bottlerocket:
+- `sources`: A Cargo workspace for first-party Rust code.
+- `packages`: A collection of Rust packages that build RPM packages.
+- `variants`: A Cargo workspace for defining what Bottlerocket variants.
+
+A new directory will be added: `kits`.
+This will be a Cargo workspace with one Cargo package for each kit.
+
+```
+kits
+├── aws
+│   ├── build.rs
+│   ├── Cargo.toml
+│   └── lib.rs
+├── core
+│   ├── build.rs
+│   ├── Cargo.toml
+│   └── lib.rs
+├── ecs
+│   ├── build.rs
+│   ├── Cargo.toml
+│   └── lib.rs
+├── k8s
+│   ├── build.rs
+│   ├── Cargo.toml
+│   └── lib.rs
+├── metal
+│   ├── build.rs
+│   ├── Cargo.toml
+│   └── lib.rs
+└── vmware
+    ├── build.rs
+    ├── Cargo.toml
+    └── lib.rs
+```
+
+A variant such as `metal-k8s-1.22` would consume the necessary kits.
+For example, it would reference `core-kit`, `k8s-kit` and `metal-kit`.
+To add `iam-authenticator`, `aws-kit` would be added.
```

Comment:
```suggestion
To add `aws-iam-authenticator`, `aws-kit` would be added.
```

As an inaugural bit of bikeshedding: `aws-iam-authenticator` might belong in the `k8s-kit` rather than `aws-kit`, since it is Kubernetes specific and can be used outside AWS - sort of like `ecr-credential-provider`.